framework/src/araui/core/ui_quick_panel.c

```diff
@@ -196,18 +199,12 @@ static void ui_quick_panel_touch_func(ui_widget_body_t *widget, ui_touch_event_t
 					if (body->state == UI_QUICK_PANEL_STATE_NONE) {
 						if (coord.y < CONFIG_UI_DISPLAY_HEIGHT / CONFIG_UI_QUICK_PANEL_APPEAR_CONST) {
 							body->state = UI_QUICK_PANEL_STATE_TRANSITION;
-							ui_widget_tween_moveto((ui_widget_t)widget,
-								0, -1 * CONFIG_UI_DISPLAY_HEIGHT,
-								CONFIG_UI_QUICK_PANEL_TRANSITION_TIME,
-								TWEEN_EASE_OUT_QUAD,
-								ui_quick_panel_disappear_tween_end_func);
+							move = ui_move_anim_create(rect.x, rect.y, 0, -1 * CONFIG_UI_DISPLAY_HEIGHT, CONFIG_UI_QUICK_PANEL_TRANSITION_TIME, UI_INTRP_EASE_OUT_QUAD);
```

Comment:
Too many branches here but it seems that can be optimized.
(I think ui_quick_panel_state_t can be expanded, you'd better use switch-case instead of if-else)

How about make api for swipe? then you can optimize ui_core_quick_panel_disappear too.

Reply:
I agree that there are so many branches.
I am considering it as next job.

Reply:
it's depend on your team's decision, not me. This is just my comment.